x/interchainqueries/module.go

```diff
@@ -76,6 +77,9 @@ func (AppModuleBasic) RegisterRESTRoutes(clientCtx client.Context, rtr *mux.Rout
 
 // RegisterGRPCGatewayRoutes registers the gRPC Gateway routes for the module.
 func (AppModuleBasic) RegisterGRPCGatewayRoutes(clientCtx client.Context, mux *runtime.ServeMux) {
+	if err := types.RegisterQueryHandlerClient(context.Background(), mux, types.NewQueryClient(clientCtx)); err != nil {
```

Comment:
could you please explain this piece if added code? why we didn't have it before? kinda complicated to understand from the func comment and innards

Reply:
this is standard code that enables REST query handlers for the module

Reply:
> there are still three attributes that are added to both event types. please consider making a func that returns common for both event types attributes and passing it to the end of the sdk.NewEvent( func via attrs...)

i think this is too much, no need to introduce an intermediate function just for a couple of attributes